common/fs/src/cache/mod.rs

```diff
@@ -849,65 +849,53 @@ where
         }
     }
 
-    /// Returns the entry that represents the supplied path or `Ok(None)`.
-    pub fn lookup(&self, path: &PathBuf, _entries: &EntryMap<T>) -> FsResult<Option<EntryKey>> {
+    /// Returns the entry that represents the supplied path.
+    /// When the path is not represented and therefore has no entry then `None` is return.
+    pub fn lookup(&self, path: &PathBuf, _entries: &EntryMap<T>) -> Option<EntryKey> {
         let mut parent = self.root;
         let mut components = into_components(path);
         // remove the first component because it will always be the root
         components.remove(0);
 
         // If the path has no components there is nothing to look up.
         if components.is_empty() {
-            return Ok(None);
+            return None;
         }
 
-        let last_component = components.pop().unwrap();
+        let last_component = components.pop()?;
 
         for component in components {
-            if let Some(entry) = self.follow_links(parent, _entries)? {
-                if let Some(parent_ref) = _entries
-                    .get(entry)
-                    .ok_or(Error::ParentLookup)?
-                    .borrow()
-                    .children()
-                    .ok_or(Error::ParentNotValid)?
-                    .get(&component)
-                {
-                    if let Some(parent_ref) = self.follow_links(*parent_ref, _entries)? {
-                        parent = parent_ref;
-                        continue;
-                    }
+            parent = self.follow_links(parent, _entries).and_then(|e| {
+                if let Some(e) = _entries.get(e) {
+                    e.borrow()
+                        .children()?
+                        .get(&component)
+                        .and_then(|entry| self.follow_links(*entry, _entries))
+                } else {
+                    error!("Failed to find entry");
```

Comment:
Is this always an error? Or should it be a warn/info?

Reply:
Good point, I'll change it to info.